### PR TITLE
[7.x~DROOLS-6810 Correct fix for potential XXE vulnerabilities

### PR DESCRIPTION
**Pull request by pibizza**
This is a better fix for the issue related to potential XXE vulnerabilities, extending #4283. 

The patch for main is in #4176.

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>

* for a <b>full downstream build</b> 
  * for <b>jenkins</b> job: please add comment: <b>Jenkins run fdb</b>
  * for <b>github actions</b> job: add the label `run_fdb`

* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Review by gitgabrio — CHANGES_REQUESTED** [review]
HI @pibizza , first of all many thanks for that PR! 
What do you think about:
1) make the "createTrasformer" a package-protected method (I think it is used only in the test)
2) if possible, to make a similar package-protected "createDocumentBilderFactory" method

What I'm driving at is that, if I'm not wrong, both "DocumentBuilderFactory" and "TransformerFactory" requires the same attributes, and it would be better to have their instantiation delegated to specific methods

(there is also some formatting to do inside DOMParserUtils.java)

Thanks Again!

**Comment by yesamer** [comment]
jenkins execute full downstream build

**Comment by pibizza** [comment]
@gitgabrio I should have implemented the suggested fixes, let me know.

**Comment by yesamer** [comment]
Jenkins execute product fdb

**Review by yesamer — APPROVED** [review]
Thank you, @pibizza LGTM. Please check the integration tests works as expected with this patch.

**Review by gitgabrio — APPROVED** [review]
Many thanks @pibizza , LGTM!

**Comment by pibizza** [comment]
Jenkins run fdb

**Comment by pibizza** [comment]
jenkins do fdb

**Comment by pibizza** [comment]
Jenkins run fdb

File: kie-dmn/kie-dmn-backend/src/main/java/org/kie/dmn/backend/marshalling/v1_1/xstream/XStreamMarshaller.java [hunk 1]
```diff
@@ -208,26 +197,7 @@ public void marshalMarshall(Object o, OutputStream out) {
...
-           serializer.setOutputProperty("{http://xml.apache.org/xslt}indent-amount", "2");
-           Source xmlSource=new SAXSource(new InputSource(new ByteArrayInputStream(xml.getBytes())));
-           StreamResult res =  new StreamResult(new ByteArrayOutputStream());            
-           serializer.transform(xmlSource, res);
-           return new String(((ByteArrayOutputStream)res.getOutputStream()).toByteArray());
-        }catch(Exception e){   
-           return xml;
-        }
-     }
+
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-backend/src/main/java/org/kie/dmn/backend/marshalling/v1_1/xstream/XStreamMarshaller.java:200]: 👍

**Comment by pibizza** [comment]
Jenkins retest this

**Comment by yesamer** [comment]
Jenkins retest this

**Comment by yesamer** [comment]
Jenkins run fdb

**Comment by yesamer** [comment]
Jenkins execute product fdb

**Comment by yesamer** [comment]
Jenkins run fdb

**Comment by yesamer** [comment]
Jenkins execute product fdb

**Comment by yesamer** [comment]
@pibizza FBD is green! :)

**Review by mariofusco — APPROVED** [review]

**Pull request merged**